File: frontends/main/src/app/configurator/services/api/field-model.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {FieldModel} from '@core/model/field-model';

@Injectable({
	providedIn: 'root'
})
export class FieldModelService {
	constructor(private http: HttpClient) {}

	getFieldModels(projectId: string): Observable<FieldModel[]> {
		return this.http.get<FieldModel[]>(`/api/superuser/configurator/projects/${projectId}/config/field-models`);
	}

	getFieldModel(projectId: string, fieldModelId: string): Observable<FieldModel> {
		return this.http.get<FieldModel>(`/api/superuser/configurator/projects/${projectId}/config/field-models/${fieldModelId}`);
	}

	createFieldModel(projectId: string, fieldModel: FieldModel): Observable<FieldModel> {
		return this.http.post<FieldModel>(`/api/superuser/configurator/projects/${projectId}/config/field-models`, fieldModel);
	}

	updateFieldModel(projectId: string, fieldModelId: string, fieldModel: FieldModel): Observable<FieldModel> {
		return this.http.put<FieldModel>(`/api/superuser/configurator/projects/${projectId}/config/field-models/${fieldModelId}`, fieldModel);
	}

	deleteFieldModel(projectId: string, fieldModelId: string): Observable<void> {
		return this.http.delete<void>(`/api/superuser/configurator/projects/${projectId}/config/field-models/${fieldModelId}`);
	}
}

File: backend/src/main/java/ch/rodano/core/services/bll/configurator/FieldModelService.java
package ch.rodano.core.services.bll.configurator;

import java.util.List;
import java.util.UUID;

import ch.rodano.api.config.FieldModelDTO;

public interface FieldModelService {

	List<FieldModelDTO> getFieldModels(UUID projectId, String view);

	default List<FieldModelDTO> getFieldModels(final UUID projectId) {
		return getFieldModels(projectId, "summary");
	}

	FieldModelDTO getFieldModel(UUID projectId, UUID fieldModelId);

	FieldModelDTO createFieldModel(UUID projectId, FieldModelDTO fieldModel);

	FieldModelDTO updateFieldModel(UUID projectId, UUID fieldModelId, FieldModelDTO fieldModel);

	void deleteFieldModel(UUID projectId, UUID fieldModelId);
}
